scripts/auth0connections/auth0connections.py

```diff
@@ -29,26 +30,44 @@ def get_client():
 
 
 def render_local_connections(config):
+    '''Returns full connection dicts, from the connections listed in the config
+    file, and rendered using the template specified.
+    
+    Uses a template specified in the connection config
+    e.g. 'template_name: hmpps-auth' in the connection configuration
+         means use the template in: ./connection_templates/hmmps-auth/
+    '''
     connections = {}
-    connections_root = Path(__file__).cwd() / Path("connections")
-    connection_dirs = [entry for entry in connections_root.iterdir() if entry.is_dir()]
-
-    for connection in connection_dirs:
-        connection_name = connection.stem
-        scripts = connection.glob("*.js")
+    template_root = Path(__file__).cwd() / Path("connection_templates")
+    template_dirs = dict((entry.stem, entry) for entry in template_root.iterdir() if entry.is_dir())
+
+    for connection_name in config:
+        connection = config[connection_name]
```

Comment:
from the little I'm understanding this was done in the following line before:

```python
scripts_rendered[name] = template.render(**config.get(connection_name))
```

but not the connection is extracted from the `config` variable here.

I guess you changed the `.get()` with `[]` to make it fail as soon as possible right?

Reply:
```
for connection_name in config:
        connection = config[connection_name]
```
We can be sure the key exists because of the previous line, but yes if that changed in future then the `[]` shows intention & failing early 👍